lib/puppet/functions/resources_deep_merge.rb

```diff
@@ -0,0 +1,86 @@
+# Deeply merge a "defaults" hash into a "resources" hash like the ones expected
+# by `create_resources()`.
+#
+# Internally calls the puppetlabs-stdlib function `deep_merge()`. In case of
+# duplicate keys the `resources` hash keys win over the `defaults` hash keys.
+#
+# Example
```

Comment:
Shouldn't this use `@example`?

Reply:
Yes, it *should*, but until puppet-strings supports it, I’m not sure.